platformio.ini

```diff
@@ -1371,6 +1371,28 @@ extra_scripts        = ${common.extra_scripts}
   buildroot/share/PlatformIO/scripts/stm32_bootloader.py
   buildroot/share/PlatformIO/scripts/mks_encrypt.py
 
+#
+# MKS Robin Nano V3
+#
+[env:mks_robin_nano_v3]
+platform      = ${common_stm32.platform}
+extends       = common_stm32
+build_flags   = ${common_stm32.build_flags} -DHAL_PCD_MODULE_ENABLED -DUSBCON -DUSBD_USE_CDC
+board         = genericSTM32F407VGT6
+board_build.core     = stm32
+board_build.variant  = MARLIN_F4x7Vx
+board_build.ldscript = ldscript.ld
+board_build.firmware = Robin_nano_v3.bin
+board_build.offset   = 0xC000
+board_upload.offset_address = 0x0800C000
+build_unflags        = ${common_stm32.build_unflags}
+debug_tool           = jlink
+upload_protocol      = jlink
+extra_scripts        = ${common.extra_scripts}
+  pre:buildroot/share/PlatformIO/scripts/generic_create_variant.py
+  buildroot/share/PlatformIO/scripts/stm32_bootloader.py
```

Comment:
Do you know why we need this script? It looks like maybe the only useful thing it does is add a couple command line arguments?

I'm wondering why we see this for lerdge and some mks boards, but no others.

Reply:
When jmz added the first MKS board on STM32, he (and I think JCNelson too), added a few generic scripts to 3 things:

1) generate variant. Instead of each board create its own "generate_variant_foo.py", now we have a generic variant script that uses the board.variant argument.

2) stm bootloader defines: this script uses the env parameters to set offset, ldscript parameters and so on. Again, instead of a script for each board (like chitu script), now we have a generic that uses env parameters to configure ldscript

3) mks_encript: all mks boards use the same encryption, but we have a few duplicated files with same content by different names... this new script is generic too.